src/routes/(app)/legacy/+page.svelte

```diff
@@ -3,19 +3,109 @@
     import Contact from "../../../components/Contact.svelte";
     import Sidebar from "../../../components/Sidebar.svelte";
     import SidebarHeader from "../../../components/SidebarHeader.svelte";
-    import {getLegacyChats} from '../../../stores/legacyChats.svelte';
+    import RespondPanel from "../../../components/RespondPanel.svelte";
+    import ChatHeader from "../../../components/ChatHeader.svelte";
+    import Loader from "../../../components/Loader.svelte";
+    import { invoke } from "@tauri-apps/api/core";
+    import { type NChat } from "../../../types/nostr";
+    import { onMount, onDestroy } from "svelte";
+    import { listen, type UnlistenFn } from "@tauri-apps/api/event";
+    import { currentIdentity } from "../../../stores/identities";
+    import ChatMessage from "../../../components/ChatMessage.svelte";
+    import { tick } from "svelte";
 
-    const legacyChats = getLegacyChats()
+    let selectedChat: string | undefined = $state(undefined);
+    let chats: NChat[] = $state([]);
+    let isLoading = $state(true);
+
+    async function getLegacyChats(): Promise<void> {
+        isLoading = true;
+        chats = [];
+        selectedChat = undefined;
+        try {
+            const fetchedChats = (await invoke("get_legacy_chats", {
+                pubkey: $currentIdentity,
+            })) as NChat;
+            const sortedChats = Object.entries(fetchedChats)
+                .sort(([, a], [, b]) => b.latest - a.latest)
+                .reduce((acc, [key, value]) => ({ ...acc, [key]: value }), {});
+            chats = sortedChats as NChat[];
```

Comment:
All the type coercion should be avoidable. Maybe wrapping the invoke call in a `fetchChats(pubkey): Promise<NChat[]>` function would work.

Reply:
Yeah, this is very ugly and definitely super optimizable. I wanted to just get something basic going to see it working, but this ultimately will be replaced with the subscription based logic so I didn't spend much time optimizing.